src/pages/AssignServerRoles/BaremetalSettings.js

```diff
@@ -0,0 +1,93 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+import React, { Component } from 'react';
+import { translate } from '../../localization/localize.js';
+import { ConfirmModal } from '../../components/Modals.js';
+import { IpV4AddressValidator } from '../../utils/InputValidators.js';
+import { ServerInputLine } from '../../components/ServerUtils.js';
+import { ActionButton } from '../../components/Buttons.js';
+
+class BaremetalSettings extends Component {
+
+  constructor(props) {
+    super(props);
+    const baremetal = this.props.model.getIn(['inputModel', 'baremetal']).toJS()
+    this.state = {
+      subnet: baremetal.subnet,
+      netmask: baremetal.netmask
+    }
+    this.origBaremetal = JSON.parse(JSON.stringify(baremetal));
+  }
+
+  handleInputChange = (e, valid, props) => {
+    let value = e.target.value;
+    if (valid) {
+      let key = props.inputName;
+      if (key === 'netmask') {
+        this.setState({netmask: value});
+      } else {
+        this.setState({subnet: value});
+      }
+    }
+  }
+
+  checkSettingsChanged = () => {
+    return (this.origBaremetal.netmask !== this.state.netmask) ||
+      (this.origBaremetal.subnet !== this.state.subnet);
+  }
+
+  cancelBaremetalSettings = () => {
+    this.setState({netmask: this.origBaremetal.netmask, subnet: this.origBaremetal.subnet});
+    this.props.cancelAction();
+  }
+
+  saveBaremetalSettings = () => {
+    let newSettings = {
+      subnet: this.state.subnet,
+      netmask: this.state.netmask
+    };
+    let model = this.props.model;
+    model = model.updateIn(['inputModel', 'baremetal'], settings => newSettings);
+    this.props.updateGlobalState('model', model);
+    this.props.cancelAction();
+  }
+
+  render() {
+    let footer = (
+      <div className='btn-row'>
+        <ActionButton type={'default'} clickAction={this.cancelBaremetalSettings}
+          displayLabel={translate('cancel')}/>
+        <ActionButton clickAction={this.saveBaremetalSettings} displayLabel={translate('save')}
+          isDisabled={!this.checkSettingsChanged()}/>
+      </div>
+    );
+    return (
+      <ConfirmModal show={this.props.show} className={'manual-discover-modal'}
+        title={translate('add.server.set.network')} onHide={this.props.cancelAction} footer={footer}>
+        <div className='description-line'>{translate('add.server.set.network.description')}</div>
+        <div className='server-details-container'>
+          <ServerInputLine label={'add.server.set.network.subnet'}
```

Comment:
It is confusing that subnet and netmask are both optional, since this would permit a user to specify a netmask without a subnet (or vice versa). I would expect them to both be required. If, on the other hand, these field are not required, it would be good to have logic to enforce that they either both be populated or both be blank.